packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx

```diff
@@ -126,123 +131,62 @@ export const PatientRegistration: React.FC<PatientRegistrationProps> = ({ savePa
-
-      // set names
-      if (patient.name.length) {
-        let name = patient.name[0];
-        patientUuidMap['preferredNameUuid'] = name.id;
-        initialFormValues.givenName = name.given[0];
-        initialFormValues.middleName = name.given[1];
-        initialFormValues.familyName = name.family;
-        if (name.given[0] === 'UNKNOWN' && name.family === 'UNKNOWN') {
-          initialFormValues.unidentifiedPatient = true;
-        }
-        if (patient.name.length > 1) {
-          name = patient.name[1];
-          patientUuidMap['additionalNameUuid'] = name.id;
-          initialFormValues.addNameInLocalLanguage = true;
-          initialFormValues.additionalGivenName = name.given[0];
-          initialFormValues.additionalMiddleName = name.given[1];
-          initialFormValues.additionalFamilyName = name.family;
-        }
-      }
-
-      initialFormValues.gender = capitalize(patient.gender);
-      initialFormValues.birthdate = patient.birthDate;
-      initialFormValues.telephoneNumber = patient.telecom ? patient.telecom[0].value : '';
-
-      patient.identifier.forEach((id) => {
-        const key = camelCase(id.system || id.type.text);
-        patientUuidMap[key] = {
-          uuid: id.id,
-          value: id.value,
-        };
-        initialFormValues[key] = id.value;
-      });
-
-      if (patient.address && patient.address[0]) {
-        const address = patient.address[0];
-        Object.keys(address).forEach((prop) => {
-          switch (prop) {
-            case 'id':
-              patientUuidMap['preferredAddressUuid'] = address[prop];
-              break;
-            case 'city':
-              initialAddressFieldValues['cityVillage'] = address[prop];
-              break;
-            case 'state':
-              initialAddressFieldValues['stateProvince'] = address[prop];
-              break;
-            case 'district':
-              initialAddressFieldValues['countyDistrict'] = address[prop];
-              break;
-            case 'extension':
-              address[prop].forEach((ext) => {
-                ext.extension.forEach((extension) => {
-                  initialAddressFieldValues[extension.url.split('#')[1]] = extension.valueString;
-                });
-              });
-              break;
-            default:
-              if (prop === 'country' || prop === 'postalCode') {
-                initialAddressFieldValues[prop] = address[prop];
-              }
-          }
-        });
-      }
-
-      if (patient.deceasedBoolean || patient.deceasedDateTime) {
-        initialFormValues.isDead = true;
-        initialFormValues.deathDate = patient.deceasedDateTime ? patient.deceasedDateTime.split('T')[0] : '';
-      }
+    if (patient) {
+      setPatientUuidMap({ ...patientUuidMap, ...getPatientUuidMapFromFhirPatient(patient) });
+      setInitialAddressFieldValues({ ...initialAddressFieldValues, ...getAddressFieldValuesFromFhirPatient(patient) });
```

Comment:
Honestly, I'm not quite sure whether all of this object merging in this file always leads to the correct form being displayed in the end.
Data gets fetched/merged all over the place. I think we should consider reworking this in the future.

For the moment, the current version should reflect the *_previous_* version of the code, just extracted into different files, i.e. I tried to not actually change (too much) behavior here.

Reply:
Yeah, personally, I think this MF has a lot of dodgy behavior. Cleaning it up would be wonderful. If we break something on the way its fine. We can then repair it.